packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts

```diff
@@ -0,0 +1,157 @@
+import { BigInt } from "@web3api/wasm-as";
+import { Big } from "as-big";
+
+import { getSushiAddress, XSUSHI_ADDRESS } from "../constants";
+import { getChainId } from "../utils/network";
+import {
+  env,
+  Ethereum_Connection,
+  Ethereum_Query,
+  Input_getTokenComponents,
+  Interface_Token,
+  Interface_TokenComponent,
+  QueryEnv,
+  Token_Query,
+  Token_Token,
+  Token_TokenType,
+} from "../w3";
+
+function getPairTokenAddresses(pairAddress: string, connection: Ethereum_Connection): string[] {
+  // get token addresses
+  const token0AddressResult = Ethereum_Query.callContractView({
+    address: pairAddress,
+    method: "function token0() external view returns (address)",
+    args: [],
+    connection: connection,
+  });
+  // if exception encountered, pair contract presumed not to exist
+  if (token0AddressResult.isErr) {
+    throw new Error("Invalid protocol token");
+  }
+  const token1AddressResult = Ethereum_Query.callContractView({
+    address: pairAddress,
+    method: "function token1() external view returns (address)",
+    args: [],
+    connection: connection,
+  });
+  if (token1AddressResult.isErr) {
+    throw new Error("Invalid protocol token");
+  }
+  return [token0AddressResult.unwrap(), token1AddressResult.unwrap()];
+}
+
+function getSushiSwapComponents(
```

Comment:
I am wondering whether we should re-use the uniswap token resolver here?

Reply:
What is the best way to do that? They are in different packages, so should we add the uniswap token resolver to npm and then add it as a dependency in the sushi token resolver??

Reply:
Sorry, I just realized I misread your original comment. They are similar enough that we could combine them into one token resolver. I think same with linkswap. The main difference is in the contract addresses being queried, and sushibar is a little different.

Reply:
Exactly, but for now, let's keep it as it is and open an issue to investigate this further.